tests/test_life_expectancy.py

```diff
@@ -33,3 +39,53 @@ def test_get_life_expectancy():
     assert sorted(rtn_full.index.to_list()) == ["F", "M"]
     assert list(rtn_full.columns.names) == ['draw', 'run']
     assert rtn_full.columns.levels[1].to_list() == [0, 1]
+
+
+def test_probability_premature_death(tmpdir, age_before_which_death_is_defined_as_premature: int = 70):
+    """
+    Test the calculation of the probability of premature death from a simulated cohort.
+
+    This function loads results from a dummy cohort (N = 100, with 37 F and 63 M) simulation where all individuals start at age 0.
+    The simulation was then run for 70 years (2010 - 2080), during which individuals could die but nobody could be born.
+    In this dummy data set, 6 F die and 23 M die prematurely, giving a probability of premature death as 0.16 and 0.37, respectively.
+    The premature deaths amongst these individuals is then the number that have died before the age of 70 (default value).
+    This test uses the calculates the probability of premature death separately for males and females using the
+    data from this simulated run and the function get_probability_of_premature_death.
+    It then compares these simulated probabilities against the total number of deaths before the age of 70 (default)
+    that occurred in the simulated cohort.
+    """
+    # load results from a dummy cohort where everyone starts at age 0.
+    target_period = (datetime.date(2010, 1, 1), datetime.date(2080, 12, 31))
+
+    results_folder_dummy_results = Path(os.path.dirname(__file__)) / 'resources' / 'probability_premature_death'
+    pickled_file = os.path.join(results_folder_dummy_results, '0', '0', 'tlo.methods.demography.pickle')
+
+    # - Compute 'manually' from raw data
+    with open(pickled_file, 'rb') as file:
+        demography_data = pickle.load(file)
+    initial_popsize = {'F':  demography_data['population']['female'][0], 'M': demography_data['population']['male'][0]}
+    deaths_total = demography_data['death'][['sex', 'age']]
+    num_premature_deaths = deaths_total.loc[deaths_total['age'] < age_before_which_death_is_defined_as_premature] \
+                                       .groupby('sex') \
+                                       .size() \
+                                       .to_dict()
+    prob_premature_death = {s: num_premature_deaths[s] / initial_popsize[s] for s in ("M", "F")}
+
+    # - Compute using utility function
+    probability_premature_death_summary = get_probability_of_premature_death(
+        results_folder=results_folder_dummy_results,
+        target_period=target_period,
+        summary=True,
+    )
+
+    # Confirm both methods gives the same answer
+    assert np.isclose(
+        probability_premature_death_summary.loc["F"].loc[(0, 'mean')],
+        prob_premature_death['F'],
+        atol=0.01
```

Comment:
@RachelMurray-Watson  I've proposed updating this test to be more direct check that the utility function gives the answer we expect. I was surprised that the tolerance allowance on this had to be 0.01 and not something smaller. Do you know why the two don't come closer?

Reply:
I'm not sure - looking at the life expectancy estimates, too, can see that there's sometimes +- 3% difference. It may be here that there's a very small population size, so the estimates from probabilities leads to inaccurate results?

Reply:
I don't think there's any stochasticity in the procedure generating these estimates though....? To my mind, the two methods in the test (manual and using the new function) should give same result as each is operating on exactly the same data?

Reply:
I was thinking that it might be something to do with the PY being used in the calc in the function, versus the just the initial cohort size in the manual calc.

eg. is the probability of premature death rightly calculated as proportion of deaths that occur at an age less than aged 70 (per manual calculation); or, is expected cumulative risk of death by age 70.

I think these two quantities are subtly different....!?

Reply:
Ah sorry, I see what you mean when applied to the same synthetic cohort...  I guess what I meant with the small cohort was that some assumptions e.g. about fractions of age groups survived may not hold as well, but I think that that does lead in to your point about PY vs the "manual" calculation as you said. 

I think that I understand the probability of premature death as a metric as the cumulative risk (like a life-table method), as you said. So do you think for a "correct" test, I should instead calculate the PY, etc. at risk etc for this cohort? 

I guess then I'm wondering what this test does tell us then; that this is an imperfect estimate?

Reply:
Ok, that's the answer then. That "small assumptions" are made about the PY spent in age-group in our calculation that mean that our manual or 'utility fn' approaches legitimately can be a little bit apart.

If that's the case, then I think we're done! I'll add in a little comment on the back of the assert line to say that.

Reply:
Amazing! Thanks so much like